### ISSUE TITLE
helloShop ideas

### ISSUE THREAD

**Issue by linuxkettle** [helloSystem/hello#169]
I'm working on a backend for a helloShop app, an easy way to manage installing and updating your apps, where you can search available apps and download them to your ~/Applications folder, but I'm not sure what the UI should look like.

If I cc @jhjacobs81, I think you'd be good here.

If you open Qt Creator (in /Applications/Developer/Qt Creator.app), launch Qt 5 Designer (in /usr/local/share/applications, sneaky little trick if you want to launch a legacy app), it comes up with a window asking you what type of form you want to make. I reccommend either Widget or Main Window, the only difference is that Main Window lets you design the menu and adds a statusbar, Widget is just an empty window. It comes up with a designer where you can design graphical interfaces, no programming neccessary. Try to avoid using anything under the KF5 sections, but if you need to, by all means do.

If you have a play around with the design, can you reply back if you find a layout you like?

Hopefully it's okay with you, I'm not sure how well Qt Creator would fit in a 800x600 screen...

**Comment by linuxkettle** [comment]
![image](https://user-images.githubusercontent.com/64388266/119070709-0e5d4480-b9d8-11eb-8571-3ffcb85dbdde.png)

**Comment by jhjacobs81** [comment]
Good morning! I will first draw up some idea’s if thats okay ;-)

**Comment by linuxkettle** [comment]
It is very not morning where I live!

Thanks for the help. I did try something but it looked like GNOME Software after a 10-magnitude earthquake!

**Comment by jhjacobs81** [comment]
its almost noon here ;) im from the Netherlands ;-) 

something i noted down before drawing:

Startpage:
Top Left: accountname and image (if connected to helloAccount)
Top Right: searchbar
Middle up: buttons: standard view /  advanched view

center screen: spotlight apps. Apps that follow the hS design guides, apps with good reviews, etc etc 

lower screen: category list

Footer: important text links

**Comment by probonopd** [comment]
> I'm working on a backend for a helloShop app, an easy way to manage installing and updating your apps

Nice, but please don't use "hello" in the name in order not to confuse people into thinking this is part of helloSystem.

> helloAccount

Such a thing is not in our plans. As per the project goals, we value privacy. App Stores are the exact opposite because they know which applications a particular user has installed. We don't want that.

In fact, I think the Mac platform went downhill around the exact time they brought the Store "back to the Mac" from iOS.

**Comment by linuxkettle** [comment]
Okay, I'm renaming it to AppShop and making it a little more FreeBSD-ish.

The original idea was just to have something like AppImageHub to take the pain out of trying to find apps that work on helloSystem.

In my opinion, it might be better as an "install it if you want" thing.

**Comment by jhjacobs81** [comment]
I disagree that a store knows what apps are installed. This is only if you want it to report. A store can be programmed to record next to nothing.

Its a single portal for users to easily find software. There’s a reason why every major platform comes with a store. Its because users want it. They dont want to spend hours surfing the web for software. 

Its rediculous not to even consider it.

**Comment by linuxkettle** [comment]
I agree, and I've moved this to <https://github.com/linuxkettle/AppShop>. I'll try to make the install process as easy as possible, and we can just see how it turns out in the long run. I will add you as a contributor, to make things a bit easier.

**Comment by linuxkettle** [comment]
From your ideas, I've come up with this:
![image](https://user-images.githubusercontent.com/64388266/119208091-2f816c00-ba90-11eb-9408-5c93b80e7cc1.png)
Each app will have a simple metadata file containing the app name, the description, the background gradient for the Featured App thing, the icon, the latest version, and other useful stuff.

**Comment by probonopd** [comment]
> I disagree that a store knows what apps are installed. This is only if you want it to report.

True. The sad truth is that the Mac App Store and the Google Play Store require an account and pretty much know what user installs what app on which machine. A concept I am heavily opposed to. It's not the operating system vendor's job to know what applications are running on users' machines imho.

**Comment by jhjacobs81** [comment]
I agree with you.

but i also firmly believe that privacy can go hand in hand with easability (is that a word?) 

i would argue that we could create a store that does not require all this data, and data for analitics could easily be anonymized. Should this ever be needed. 

Same goes for accounts. They should only be used for easily syncing apps and preferences. Required data could be stored encrypted, with noone other then the user have access to this data. 

This is the idea behind what Apple does with their appstore and icloud accounts. It allows for easily synchronisation across multiple accounts. If not for the privacy issues, they have done this right! 

If we could create the same experience, but without all the data mining.. well i think we suddenly have a lot more users :) and users are willing to pay for that. You’d find yourself working on helloSystem fulltime with the right funding.

**Comment by probonopd** [comment]
The first 20 or so years of the Macintosh you didn't need an account. I would like to keep it that way. 

>  They should only be used for easily syncing apps and preferences.

Connect computer A to computer B, copy over the applications and preferences, done.

**Comment by jhjacobs81** [comment]
I’d never said you would NEED an acount, you are not reading what i’m saying. 

In any case, this distro clearly isnt for me :) i wish you good luck though! I’ll keep a sideway look to see how things progress :)

**Comment by linuxkettle** [comment]
For a while I had a private git repo for that, I set up an init.d script to sync the changes, and it synced /home, /var/lib/flatpak and /var/snap.

It can be done. It just isn't easy.

But what we *could* do, it set up an app that runs a node server on one machine and a bunch of clients on the others to sync your data, and run it on the LAN network. That way, the data gets to all of your devices without leaving your house.

I believe I published a repository called Chatter, it's a re-inactment of an old school project that no teacher knew about and I recreated it with node to remind my friends.
